Add environment for zwicky (Fullerton)

geoffrey4444 opened the pull request:
## Proposed changes

Add an environment file to build spectre on zwicky.

Note that because zwicky is so old, I have to build with an old version of libxsmm (1.5.2, newest is 1.9).

All tests pass, except test 373 fails (some answers differ from the expected answers by roundoff).

Is the failing test a cause for concern? Are we sure the comparisons should always be exact, instead of differing by roundoff?

<!--
At a high level, describe what this PR does.
-->

### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [x] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

kidder commented on docs/Installation/InstallationOnClusters.md:
```diff
@@ -61,3 +61,15 @@ Follow the general instructions using `wheeler` for `SYSTEM_TO_RUN_ON`, except
+you do not need to install any dependencies, so you can skip steps 5 and 6. 
+Only gcc is supported (use `zwicky_gcc.sh`).
+
+Note that this is a very old machine, so you'll need to load many module to 
```
module -> modules

kidder commented on support/Environments/zwicky_gcc.sh:
```diff
@@ -24,12 +24,13 @@ module unload brigand-master-gcc-7.3.0-7mqn34k
-module unload libxsmm-1.5.2-gcc-7.3.0-5zli3hq
+module unload libxsmm/master
 module unload curl-7.60.0-gcc-7.3.0-dbnl246
 module unload openssl-1.0.2n-gcc-7.3.0-mtsyoqu
```
openssl/master

kidder replied:
and I assume you want to squash the two commits as well

wthrowe commented:
Test 373 is Unit.Numerical.LinearOperators.ApplyMatrices?  (Quote the test names; the numbers don't mean anything and change as new tests ar added.)

I think all the exact comparisons in that test should be such, but I'm not certain of one set.  I would need test output to comment further.

geoffrey4444 commented:
Thanks, @wthrowe. Here's the test output:

`-bash-3.2$ ./bin/RunTests Unit.Numerical.LinearOperators.ApplyMatrices
--------------------------------------------------------------------------
WARNING: There is at least non-excluded one OpenFabrics device found,
but there are no active ports detected (or Open MPI was unable to use
them).  This is most certainly not what you wanted.  Check your
cables, subnet manager configuration, etc.  The openib BTL will be
ignored for this job.

  Local host: zwicky
--------------------------------------------------------------------------
Charm++> Running on MPI version: 3.1
Charm++> level of thread support used: MPI_THREAD_FUNNELED (desired: MPI_THREAD_FUNNELED)
Charm++> Running in SMP mode: numNodes 1,  1 worker threads per process
Charm++> The comm. thread both sends and receives messages
Converse/Charm++ Commit ID: v6.8.0-0-ga36028edb
Warning> Randomization of virtual memory (ASLR) is turned on in the kernel, thread migration may not work! Run 'echo 0 > /proc/sys/kernel/randomize_va_space' as root to disable it, or try running with '+isomalloc_sync'.
CharmLB> Load balancer assumes all CPUs are same.
Charm++> Running on 1 unique compute nodes (12-way SMP).
Charm++> cpu topology info is gathered in 0.001 seconds.
SpECTRE Build Information:
Version:                      0.0.0
Compiled on host:             zwicky.fullerton.edu
Compiled in directory:        /home/geoffrey/Codes/spectre/build
Source directory is:          /home/geoffrey/Codes/spectre
Compiled on git branch:       zwicky-environment
Compiled with git hash:       d272f441d2847e046f80baeae6b2232dc2165481
Linked on:                    Sun Jun 17 11:02:53 2018

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
RunTests is a Catch v2.2.1 host application.
Run with -? for options

-------------------------------------------------------------------------------
Unit.Numerical.LinearOperators.ApplyMatrices
-------------------------------------------------------------------------------
/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:139
...............................................................................

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,2)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,2)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,2)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,2)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,-1,-2.77556e-17,1)
  ==
  (1,0,-1,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,2)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,-1,-2.77556e-17,1)
  ==
  (1,0,-1,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,2)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,-1,-2.77556e-17,1)
  ==
  (1,0,-1,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,2)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,-1,-2.77556e-17,1)
  ==
  (1,0,-1,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,2)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,3)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,3)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,0,0,0,-1,-2.77556e-17,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,0,0,0,-1,-2.77556e-17,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1,-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,4)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1,-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,4)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,0.447214,1.24127e-17,-0.447214,-0.447214,-1.24127e-17,0.
  447214,-1,-2.77556e-17,1)
  ==
  (1,0,-1,0.447214,0,-0.447214,-0.447214,0,0.447214,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,4)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,0.447214,1.24127e-17,-0.447214,-0.447214,-1.24127e-17,0.
  447214,-1,-2.77556e-17,1)
  ==
  (1,0,-1,0.447214,0,-0.447214,-0.447214,0,0.447214,-1,0,1)
with messages:
  *source_extents := (4,2)
  *dest_extents := (3,4)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,2)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,2)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,-1,-2.77556e-17,1)
  ==
  (1,0,-1,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,2)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,-1,-2.77556e-17,1)
  ==
  (1,0,-1,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,2)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,3)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,3)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,3)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,3)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,0,0,0,-1,-2.77556e-17,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,0,0,0,-1,-2.77556e-17,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,0,0,0,-1,-2.77556e-17,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,0,0,0,-1,-2.77556e-17,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,0,0,0,-1,-2.77556e-17,1)
  ==
  (-1,0,1,0,0,0,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,3)
  *powers := (1,2)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,0,0,0,-1,-2.77556e-17,1)
  ==
  (-1,0,1,0,0,0,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,3)
  *powers := (1,2)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,0,0,0,-1,-2.77556e-17,1)
  ==
  (-1,0,1,0,0,0,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,3)
  *powers := (1,2)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,0,0,0,-1,-2.77556e-17,1)
  ==
  (-1,0,1,0,0,0,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,3)
  *powers := (1,2)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1,-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,4)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1,-1,-2.77556e-17,1)
  ==
  (-1,0,1,-1,0,1,-1,0,1,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,4)
  *powers := (1,0)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,0.447214,1.24127e-17,-0.447214,-0.447214,-1.24127e-17,0.
  447214,-1,-2.77556e-17,1)
  ==
  (1,0,-1,0.447214,0,-0.447214,-0.447214,0,0.447214,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,4)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,2.77556e-17,-1,0.447214,1.24127e-17,-0.447214,-0.447214,-1.24127e-17,0.
  447214,-1,-2.77556e-17,1)
  ==
  (1,0,-1,0.447214,0,-0.447214,-0.447214,0,0.447214,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,4)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-0.2,-5.55112e-18,0.2,-0.2,-5.55112e-18,0.2,-1,-2.77556e-
  17,1)
  ==
  (-1,0,1,-0.2,0,0.2,-0.2,0,0.2,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,4)
  *powers := (1,2)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-2.77556e-17,1,-0.2,-5.55112e-18,0.2,-0.2,-5.55112e-18,0.2,-1,-2.77556e-
  17,1)
  ==
  (-1,0,1,-0.2,0,0.2,-0.2,0,0.2,-1,0,1)
with messages:
  *source_extents := (4,3)
  *dest_extents := (3,4)
  *powers := (1,2)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-2.77556e-17,-2.77556e-17,1,1)
  ==
  (-1,-1,0,0,1,1)
with messages:
  *source_extents := (2,4)
  *dest_extents := (2,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-2.77556e-17,-2.77556e-17,1,1)
  ==
  (-1,-1,0,0,1,1)
with messages:
  *source_extents := (2,4)
  *dest_extents := (2,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-2.77556e-17,-2.77556e-17,1,1)
  ==
  (-1,-1,0,0,1,1)
with messages:
  *source_extents := (2,4)
  *dest_extents := (2,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-2.77556e-17,-2.77556e-17,1,1)
  ==
  (-1,-1,0,0,1,1)
with messages:
  *source_extents := (2,4)
  *dest_extents := (2,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,-1,2.77556e-17,-2.77556e-17,-1,1)
  ==
  (1,-1,0,0,-1,1)
with messages:
  *source_extents := (2,4)
  *dest_extents := (2,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,-1,2.77556e-17,-2.77556e-17,-1,1)
  ==
  (1,-1,0,0,-1,1)
with messages:
  *source_extents := (2,4)
  *dest_extents := (2,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,-1,2.77556e-17,-2.77556e-17,-1,1)
  ==
  (1,-1,0,0,-1,1)
with messages:
  *source_extents := (2,4)
  *dest_extents := (2,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,-1,2.77556e-17,-2.77556e-17,-1,1)
  ==
  (1,-1,0,0,-1,1)
with messages:
  *source_extents := (2,4)
  *dest_extents := (2,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-1,-2.77556e-17,-2.77556e-17,-2.77556e-17,1,1,1)
  ==
  (-1,-1,-1,0,0,0,1,1,1)
with messages:
  *source_extents := (2,4)
  *dest_extents := (3,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-1,-2.77556e-17,-2.77556e-17,-2.77556e-17,1,1,1)
  ==
  (-1,-1,-1,0,0,0,1,1,1)
with messages:
  *source_extents := (2,4)
  *dest_extents := (3,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,0,-1,2.77556e-17,0,-2.77556e-17,-1,0,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (2,4)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,0,-1,2.77556e-17,0,-2.77556e-17,-1,0,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (2,4)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-2.77556e-17,-2.77556e-17,1,1)
  ==
  (-1,-1,0,0,1,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (2,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-2.77556e-17,-2.77556e-17,1,1)
  ==
  (-1,-1,0,0,1,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (2,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,-1,2.77556e-17,-2.77556e-17,-1,1)
  ==
  (1,-1,0,0,-1,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (2,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,-1,2.77556e-17,-2.77556e-17,-1,1)
  ==
  (1,-1,0,0,-1,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (2,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-1,-2.77556e-17,-2.77556e-17,-2.77556e-17,1,1,1)
  ==
  (-1,-1,-1,0,0,0,1,1,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (3,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-1,-2.77556e-17,-2.77556e-17,-2.77556e-17,1,1,1)
  ==
  (-1,-1,-1,0,0,0,1,1,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (3,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-1,-2.77556e-17,-2.77556e-17,-2.77556e-17,1,1,1)
  ==
  (-1,-1,-1,0,0,0,1,1,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (3,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-1,-2.77556e-17,-2.77556e-17,-2.77556e-17,1,1,1)
  ==
  (-1,-1,-1,0,0,0,1,1,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (3,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,0,-1,2.77556e-17,0,-2.77556e-17,-1,0,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,0,-1,2.77556e-17,0,-2.77556e-17,-1,0,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,0,-1,2.77556e-17,0,-2.77556e-17,-1,0,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,0,-1,2.77556e-17,0,-2.77556e-17,-1,0,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,0,-1,-2.77556e-17,0,-2.77556e-17,1,0,1)
  ==
  (-1,0,-1,0,0,0,1,0,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (3,3)
  *powers := (2,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,0,-1,-2.77556e-17,0,-2.77556e-17,1,0,1)
  ==
  (-1,0,-1,0,0,0,1,0,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (3,3)
  *powers := (2,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,0,-1,-2.77556e-17,0,-2.77556e-17,1,0,1)
  ==
  (-1,0,-1,0,0,0,1,0,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (3,3)
  *powers := (2,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,0,-1,-2.77556e-17,0,-2.77556e-17,1,0,1)
  ==
  (-1,0,-1,0,0,0,1,0,1)
with messages:
  *source_extents := (3,4)
  *dest_extents := (3,3)
  *powers := (2,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-2.77556e-17,-2.77556e-17,1,1)
  ==
  (-1,-1,0,0,1,1)
with messages:
  *source_extents := (4,4)
  *dest_extents := (2,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-2.77556e-17,-2.77556e-17,1,1)
  ==
  (-1,-1,0,0,1,1)
with messages:
  *source_extents := (4,4)
  *dest_extents := (2,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,-1,2.77556e-17,-2.77556e-17,-1,1)
  ==
  (1,-1,0,0,-1,1)
with messages:
  *source_extents := (4,4)
  *dest_extents := (2,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,-1,2.77556e-17,-2.77556e-17,-1,1)
  ==
  (1,-1,0,0,-1,1)
with messages:
  *source_extents := (4,4)
  *dest_extents := (2,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-1,-2.77556e-17,-2.77556e-17,-2.77556e-17,1,1,1)
  ==
  (-1,-1,-1,0,0,0,1,1,1)
with messages:
  *source_extents := (4,4)
  *dest_extents := (3,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,-1,-1,-2.77556e-17,-2.77556e-17,-2.77556e-17,1,1,1)
  ==
  (-1,-1,-1,0,0,0,1,1,1)
with messages:
  *source_extents := (4,4)
  *dest_extents := (3,3)
  *powers := (0,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,0,-1,2.77556e-17,0,-2.77556e-17,-1,0,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (4,4)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (1,0,-1,2.77556e-17,0,-2.77556e-17,-1,0,1)
  ==
  (1,0,-1,0,0,0,-1,0,1)
with messages:
  *source_extents := (4,4)
  *dest_extents := (3,3)
  *powers := (1,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:97: FAILED:
  CHECK( apply_matrices(matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,8.32667e-17,-1,-2.77556e-17,3.08149e-33,-2.77556e-17,1,-8.32667e-17,1)
  ==
  (-1,8.32667e-17,-1,0,0,0,1,-8.32667e-17,1)
with messages:
  *source_extents := (4,4)
  *dest_extents := (3,3)
  *powers := (2,1)

/home/geoffrey/Codes/spectre/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp:99: FAILED:
  CHECK( apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)), source_extents) == get(get<ScalarTag>(result)) )
with expansion:
  (-1,8.32667e-17,-1,-2.77556e-17,3.08149e-33,-2.77556e-17,1,-8.32667e-17,1)
  ==
  (-1,8.32667e-17,-1,0,0,0,1,-8.32667e-17,1)
with messages:
  *source_extents := (4,4)
  *dest_extents := (3,3)
  *powers := (2,1)`

wthrowe commented:
OK, I think that's fine.  I made an incorrect assumption about what should be identical.  Here's my proposed change to the test.  Could you try it and let me know if it fixes your issue?
```patch
From b8a539e6627cc80a6912df9a5c8dfd8e29367245 Mon Sep 17 00:00:00 2001
From: William Throwe <wtt6@cornell.edu>
Date: Sun, 17 Jun 2018 19:18:09 -0400
Subject: Allow error between Variables and DV matrix apply

---
 .../NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp b/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp
index 55a3eb1d7..f5c963ecd 100644
--- a/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp
+++ b/tests/Unit/NumericalAlgorithms/LinearOperators/Test_ApplyMatrices.cpp
@@ -93,10 +93,14 @@ struct CheckApply<Dim, Dim> {
     const auto ref_matrices =
         make_array<std::reference_wrapper<const Matrix>, Dim>(matrices);
     CHECK(apply_matrices(ref_matrices, source_data, source_extents) == result);
-    CHECK(apply_matrices(matrices, get(get<ScalarTag>(source_data)),
-                         source_extents) == get(get<ScalarTag>(result)));
+    const auto datavector_result =
+        apply_matrices(matrices, get(get<ScalarTag>(source_data)),
+                       source_extents);
+    for (const auto& p : datavector_result - get(get<ScalarTag>(expected))) {
+      CHECK(approx(p) == 0.);
+    }
     CHECK(apply_matrices(ref_matrices, get(get<ScalarTag>(source_data)),
-                         source_extents) == get(get<ScalarTag>(result)));
+                         source_extents) == datavector_result);
   }
 };
 
-- 
2.16.1
```

geoffrey4444 commented:
Thanks again, @wthrowe! With those changes, the test passes.

geoffrey4444 commented:
OK, I rebased and included @wthrowe's fix to the ApplyMatrices test. Please also note that I fixed whitespace at ends of lines in the orca installation notes, which no one noticed before pushing to master. (Editing docs in the github browser seems to bypass the guards against this).

kidder commented on support/Environments/zwicky_gcc.sh:
```diff
@@ -0,0 +1,72 @@
+module unload jemalloc-4.5.0-gcc-7.3.0-adzurmv
+module unload libxsmm/master
+module unload curl-7.60.0-gcc-7.3.0-dbnl246
+module unload openssl-1.0.2n-gcc-7.3.0-mtsyoqu
```
doesn't match what you load

geoffrey4444 committed:
Fix Test_ApplyMatrices to not assume exact for approximate comparison

geoffrey4444 committed:
Add support for zwicky (Fullerton)

kidder commented:
@wthrowe please look at (and merge if you approve)
